### PR TITLE
Refactor and run

### PR DESCRIPTION
**Pull request by bretttjohnson1**
Refactor the code into sub directories.
Changed the way code is run from to combine sending and receiving requests into one script that takes comands

### CODE REVIEW

File: syncr_backend/run_backend.py [hunk 1]
```diff
@@ -1,31 +1,79 @@
...
+from syncr_backend.network.listen_requests import listen_requests
 
 
-def execute_node_function(function_name: str, args: List[str]):
+def run_backend() -> None:
     """
-    Runs a function with the given args
+    Runs the backend
+    """
+    input_args_parser = argparse.ArgumentParser()
```
Comment (matthewbentley) [inline syncr_backend/run_backend.py]: YES :D

File: syncr_backend/run_backend.py [hunk 2]
```diff
@@ -1,31 +1,79 @@
...
+        "ip",
+        type=str,
+        nargs=1,
+        help='ip to bind listening server',
+    )
+    input_args_parser.add_argument(
+        "port",
+        type=str,
+        nargs=1,
+        help='ip to bind listening server',
```
Comment (matthewbentley) [inline syncr_backend/run_backend.py]: s/ip/port/

File: syncr_backend/run_backend.py [hunk 3]
```diff
@@ -1,31 +1,79 @@
...
+    )
+    arguments = input_args_parser.parse_args()
 
-    :param function_name: string name of the function to run
-    :param args: arguments for the function to run
+    request_listen_thread = threading.Thread(
```
Comment (matthewbentley) [inline syncr_backend/run_backend.py]: Can we add an optional `mode` option, one for interactive and one for daemon? And do this for interactive and for daemon just run the server

File: syncr_backend/run_backend.py [hunk 4]
```diff
@@ -1,31 +1,79 @@
...
+    request_listen_thread.start()
+    read_cmds_from_cmdline()
+    request_listen_thread.join()
+
+
+def read_cmds_from_cmdline() -> None:
     """
+    Read and execute commands given as cmdline input
+    """
+    exit = False
```
Comment (matthewbentley) [inline syncr_backend/run_backend.py]: `exit` is a built in function, rename this

File: syncr_backend/run_backend.py [hunk 5]
```diff
@@ -1,31 +1,79 @@
...
+    while not exit:
+        command = input("5yncr >>> ").split(' ')
+        return_list = ["", []]
+        parse_cmd_thread = threading.Thread(
+            target=parse_cmd,
+            args=[command, return_list],
+        )
+        parse_cmd_thread.start()
```
Comment (matthewbentley) [inline syncr_backend/run_backend.py]: Why do thread.start followed by thread.join?

File: syncr_backend/run_backend.py [hunk 6]
```diff
@@ -40,12 +88,80 @@ def main():
...
+            found_function = True
+        elif function_name == "send_chunk_request":
+            print(send_requests.send_chunk_request(
+                args[0],
+                int(args[1]),
+                args[2],
+                int(args[4]),
+            ))
+            found_function = True
+    # placeholder only for debugging
```
Comment (matthewbentley) [inline syncr_backend/run_backend.py]: Then get rid of it for merging?

**Comment by matthewbentley** [comment]
Also, you may need to make changes to the tracker for the new modules

File: syncr_backend/run_backend.py (hunk 6 above)
Comment (bretttjohnson1) [reply syncr_backend/run_backend.py]: the placeholder is when we actually begin testing node interaction, I wanted to make it more convenient to test out specific functions.

File: syncr_backend/run_backend.py (hunk 5 above)
Comment (bretttjohnson1) [reply syncr_backend/run_backend.py]: In a future commit, we will have a more elegant way to exit the program. Should an exit signal be sent to the command thread and the backend thead, we do not want to have the command thread read it first and exit then terminating the program while the child process is alive. The thread.join is to ensure that both threads come to an end before the program ends.

**Pull request merged**